### PR TITLE
fix: path of stacked area series with missing values

### LINKED ISSUE #680: Missing values default to value when stacked
**Describe the bug**
When a value is missing from a given series, the value is correctly skipped when the series is not stacked. But when the values are stacked the charts fills in a value.

See https://codesandbox.io/s/confident-https-gwrks?file=/src/App.tsx

**Expected behavior**
Should skip the value and not loop back.

**Screenshots**
image

> Area opacity set to zero for clarity

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

Fixes #680

Accounts for sorted `xValues` when sorting filled missing values.

### Checklist

Delete any items that are not applicable to this PR.

- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

File: browsers/browsers.test.ts [hunk 1]
```diff
@@ -60,6 +60,15 @@ describe('smoke tests', () => {
   });
 
   afterAll(async() => {
+    const entries = await driver.manage().logs().get(logging.Type.BROWSER);
+    console.log('EEEERRORS', JSON.stringify(entries));
```
Comment (markov00) [inline browsers/browsers.test.ts:64]: MMMMMM ppppprrrrooobbbbabbblyyy wwwweeee shhhhhooouuuuulddd fiiiix thiiiiiis tttttttypoooooo :D

Comment (nickofthyme) [reply browsers/browsers.test.ts:64]: lol, see #705

Comment (nickofthyme) [reply browsers/browsers.test.ts]: removed see [force push](https://github.com/elastic/elastic-charts/compare/6063029050ad3107dafca1aee56b24a4ff38be18..c5490c2a24d2976aa12ad9e6d3b4a22c2ae48b98)

**Comment by nickofthyme** [comment]
@markov00 yeah I first thought that screenshot was broken but then I realized I fixed it 😍

**Pull request merged**